magento/pwa-studio
# fix: no more infinite error loop if cart creation fails

## Description

Cart creation is currently handled by two actions. `getCartDetails` which calls `createCart` if there is no cart id when the app initializes. Unfortunately, if there is an error during the `createCart` action, we only dispatched the error (which reset the cart slice to initial state where `cartId === null`) and then retried the `getCartDetails` action. This results in an infinite loop.

The fix is to allow `createCart` to throw, but to gracefully handle the error and just return from `getCartDetails` which allows the user to continue browsing the site or doing whatever else they want, so long as it doesn't involve a cart.

I had considered popping a toast for this, but that gets into territory I'd rather leave for later. We should _always_ have a valid cart when we try to perform some action that requires one, but we shouldn't prevent a user from browsing the site if creation failed.

## Related Issue





Closes [PWA-732](https://jira.corp.magento.com/browse/PWA-732).
Closes #2522 

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. In chrome dev tools, block the /graphql endpoint.
2. Clear cache and refresh the page.
3. You might see a lot of network failures, but it should not infinitely request.

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
